source/plugin/data/custom/datamanipulators.rst

```diff
@@ -0,0 +1,62 @@
+=============================
+Implementing DataManipulators
+=============================
+
+.. javadoc-import::
+    org.spongepowered.api.data.DataSerializable
+    org.spongepowered.api.data.DataHolder
+    org.spongepowered.api.data.manipulator.DataManipulator
+    org.spongepowered.api.data.manipulator.ImmutableDataManipulator
+    org.spongepowered.api.data.manipulator.mutable.common.AbstractBoundedComparableData
+    org.spongepowered.api.data.manipulator.mutable.common.AbstractData
+    org.spongepowered.api.data.manipulator.mutable.common.AbstractSingleData
+    org.spongepowered.api.data.value.ValueContainer
+    org.spongepowered.api.inventory.ItemStack
+
+
+A core part of custom data is the :javadoc:`DataManipulator`. To implement it, the first thing you'll want to decide is
+whether you'll want to create a separate API or not for your custom data. Generally speaking it's best to separate the
+API from the implementation (and this is what SpongeAPI does), but if it won't be seen by other developers than you can
+just put both the same class.
+
+You'll want to define an API method for each "unit" your data, such as a ``String``, ``int``,
+:javadoc:`ItemStack` or a custom type like ``Home``. These units will be wrapped in a :doc:`Value <values>`, which will
+allow it to be accessed with :doc:`Keys <../keys>`.
+
+.. TODO: Code example of a value getter
+
+Now, pick which of the :javadoc:`AbstractData` types you'll extend from. While you could implement from
+scratch, these abstract types remove a *lot* of the work that needs to be done implementing :javadoc:`ValueContainer`
+methods. A full list can be found in :javadoc:`org.spongepowered.api.data.manipulator.mutable.common`.
+
+You need to create two different classes - one which is mutable and implements :javadoc:`DataManipulator` and your
+abstract type, and an immutable version which implements :javadoc:`ImmutableDataManipulator` and your *immutable*
+abstract type.
+
+For all types, you'll need to define the :javadoc:`DataManipulator#asImmutable()`/``asMutable()`` methods on
+each as well - this is as simple as copying the existing objects into a constructor for the alternate version.
+
+You should also implement :javadoc:`DataSerializable#toContainer()`. This will allow your data to be 
+:doc:`serialized <../serialization>` to :javadoc:`DataHolder`\ s or written to config files.
```

Comment:
Should make a note about how important it is to use `super.toContainer()` for `version` numbers.